rdbms/src/main/java/io/spine/server/storage/jdbc/TypeMapping.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2017, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.server.storage.jdbc;
+
+/**
+ * A {@link Type}-to-name mapping.
+ *
+ * <p>A data type may have different names in different databases.
+ * E.g. for binary data {@code BLOB} is used in MySQL, but in PostgreSQL it's {@code BYTEA}.
+ *
+ * <p>A type mapping provides a flexible way to point out
+ * database specific names of {@linkplain Type types}.
+ *
+ * @author Dmytro Grankin
+ */
+public interface TypeMapping {
+
+    /**
+     * Obtains the name of the specified {@link Type}.
+     *
+     * @param type the type to get the name
+     * @return the type name
+     * @throws IllegalStateException if the name for the specified type is not defined
+     */
+    String getTypeName(Type type);
```

Comment:
To make it looking a non-getter, I would call it something like `typeNameFor(Type)`. And it should return `TypeName`. Not `String`.
That's just an impression from this public API element.

Reply:
Done.